### ISSUE TITLE
why tf.nn.relu(net_prob_mat)  not tf.nn.sigmoid(net_prob_mat) ?

### ISSUE THREAD

**Issue by onlyNata** [skokec/segdec-net-jim2019#18]
Thank for your great work !

segdec_model.py line 118 and 120:
```
net_prob_mat = layers.conv2d(net, 1, [1, 1], scope='conv5', activation_fn=None)
decision_net = self.decision_net_fn(net, tf.nn.relu(net_prob_mat))
```

I'm confused.  in my opinion,the output layer mybe not use BN(just like the line 140, normalizer_fn=None).
And, reuse the SegModel result，mybe tf.nn.sigmoid(net_prob_mat) or tf.nn.softmax(net_prob_mat)(for multi-classifica)，but why Relu?
 
Looking forward to your reply，thanks~

**Comment by skokec** [comment]
Hi,

you are correct about the batch norm in line 118, output layer should normally not have BN. But, I am not sure how much negative influence it had on cross-entropy loss. Let me know if you will be getting better results without batchnorm.

But, be careful that you do use batchnorm and relu when passing `net_prob_mat` to `decision_net_fn`, as I think both batch norm and relu are needed since this gets concatenated with the features from segmentation network. Those features do have batchnorm and relu applied and are therefore in specific range. If `net_prob_mat` would be in different range due to not being normalized then weights would be in different ranges resulting in poor gradients for `net_prob_mat` or for segmentation features. Maybe, ReLU could be removed since we do not pass gradients back through `net_prob_mat` and non-linearity is not needed. 

Best,
Domen

**Comment by onlyNata** [comment]
Thanks for your reply.
now，i am  clear .
